### ISSUE TITLE
Language support: C# and C++

### ISSUE THREAD

**Issue by sor** [lppedd/idea-return-highlighter#12]
Hey, I really like the idea of your plugin and thought it would be nice to have for C# (Rider) and maybe C++ (CLion) as well.
Additionally you could highlight **break**, **continue**, **throw** and **goto** statements, then all of the possibly hidden control flow statements would be nicely visible.

**Comment by lppedd** [comment]
I think we should stop at `return` and, maybe, `goto` for the languages that support it.

I'd like to avoid cluttering the gutter icon space. Anyway I'll certainly look into C++ during the weekend.

Can you split the two languages in two issues?

**Comment by sor** [comment]
The solution with the gutter is only one possible (for you the easiest since you already did this)
I was thinking to

**Comment by lppedd** [comment]
@sor So basically what I could do is
```
return    > gutter icon
throw     > gutter icon 
goto      > different foreground / background
break     > different foreground / background
continue  > different foreground / background
```

Is it ok?
I plan on implement return only for now, I will have more time for the other in the beginning of January.

**Comment by sor** [comment]
That's super nice, thanks!
For the far future it would be nice to be able to configure to either use colors and/or an icon.

**Comment by lppedd** [comment]
@sor so do you prefer I go C# first?

**Comment by sor** [comment]
Yeah, C++ was only nice-to-have for me since I'm coding quite a lot of C#
atm.

Edoardo Luppi <notifications@github.com> schrieb am Do., 19. Dez. 2019,
19:53:

**Comment by Rody66** [comment]
> @sor so do you prefer I go C# first?

@lppedd, Thanks in advice for C#!

**Comment by lppedd** [comment]
I was studying how to do this.  
It seems there is no straightforward way, as C# is not available as a plugin.  
I'll have to download Rider and check where the PSI elements are stored
